archive/zstd_test.cpp

```diff
@@ -0,0 +1,139 @@
+// SPDX-FileCopyrightText: 2024 David Zero <zero-one@zer0-one.net>
+//
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include "archive/zstd.h"
+
+#include "etest/etest2.h"
+
+#include <tl/expected.hpp>
+
+#include <cstdint>
+#include <span>
+#include <string>
+#include <vector>
+
+int main() {
+    etest::Suite s{"zstd"};
+
+    using namespace archive;
+
+    s.add_test("trivial decode", [](etest::IActions &a) {
+        // NOLINTNEXTLINE(modernize-avoid-c-arrays)
```

Comment:
Why not std::array? Constructing one like `constexpr auto compress = std::to_array<std::uint8_t>({0x28, 0xb5, ...});` wouldn't add more code and would have the same runtime speed.